ISSUE-105: Transaction Isolation Level

aksm opened the pull request:
Resolves #105.

Confirmed that the esmero-db container comes up without issue for both ARM and AMD/Intel and running a mysql/mariadb query shows the right transaction isolation level.

Commit:
Add --transaction-isolation=READ-COMMITTED option for mysql and mariadb container commands.

DiegoPino commented:
@aksm since you are here, can you check mariadb logs? I think I saw a notice about --binlog-expire-logs-seconds=172800 (which is not enabled by default?) maybe we don't need that?

DiegoPino commented:
I also have this on a production instance (mariadb)

--default-authentication-plugin=mysql_native_password  --max_allowed_packet=256M --sort_buffer_size=512K --lower_case_table_names=1 --innodb_read_only_compressed=OFF

Maybe we need --log-bin=ROW for the -binlog-expire? part

DiegoPino commented:
See
```SHELL
2023-02-23 15:42:58 0 [Warning] You need to use --log-bin to make --expire-logs-days or --binlog-expire-logs-seconds work.
```

DiegoPino commented:
Related to 
https://tracker.moodle.org/browse/MDL-72131

For some reason this was an issue ... (warning on the logs) after updating to mariadb:10.6.12-focal

Commit:
Remove --default-authentication-plugin (deprecated in mysql and warning in mariadb) option and add --log-bin=mysqld-bin in accordance with warning about --binlog-expire-logs-seconds.

aksm commented:
@DiegoPino I added the `--log-bin=mysqld-bin` option in accordance with the warning since simply removing it on existing installations is not enough. Will do some testing on a fresh deployment to see if it's ok to remove `--binlog-expire-logs-seconds` option entirely, but until then ok to go with this for now?

I also removed the `--default-authentication-plugin` option since it's deprecated in mysql and gives a warning in mariadb.

DiegoPino commented:
Hi, please confirm with a fresh deployment that all this works. Drupal is quite weird re: db connection auth. Thanks so much

aksm commented:
@DiegoPino Likewise confirmed here to be working in new deployments on ARM64 and AMD/Intel. Thanks!
